feat: single connection to either mqtt service or core mqtt (#3862)
thin-edge/thin-edge.io: docs/src/operate/c8y/connect-mqtt-service.md

Author: Resolved by [3b63cc9](https://github.com/thin-edge/thin-edge.io/pull/3862/commits/3b63cc97afd4db4804adedcf1eab733d3cfcbdfc)

Author: Actually, during this transition phase when both endpoints are available, this comment is still valid. So, I haven't touched that note.

## MQTT Service

%%te%%, when connected to Cumulocity, connects to its [Core MQTT](https://cumulocity.com/docs/device-integration/mqtt/) 
endpoint by default, which only supports a predefined set of topics that the device can publish data to

> and receive data from in predefined data formats (e.g: SmartREST over JSON over MQTT).

Author: Resolved by [e772f1c](https://github.com/thin-edge/thin-edge.io/pull/3862/commits/e772f1c4d3ed8fdcd4feb337761db88c2e94be73)

>    This step establishes the bridge connection to both the core endpoint as well as the mqtt service endpoints simultaneously.

Reviewer: ```suggestion
   This step establishes the bridge connection to  the mqtt service endpoints instead of the core endpoint.
```

1. Once connected, all messages published to `c8y/custom/out/#` topics are forwarded to the MQTT service endpoint,
   without the `c8y/custom/out/` prefix.

   ```sh
   tedge mqtt pub c8y/custom/out/test/topic '{ "hello": "world" }'
   ```

   The receipt of the published message on `test/topic` can be validated on Cumulocity.
2. Similarly any messages published to the subscribed `demo/topic` on Cumulocity

>    are published to the corresponding local bridge topic with a `c8y/custom/in/` topic prefix.

Author: I'll take another attempt using some realistic examples for inbound and outbound topics using a custom topic scheme.

>    are published to the corresponding local bridge topic with a `c8y/custom/in/` topic prefix.

Author: Refactored the existing doc in [3b63cc9](https://github.com/thin-edge/thin-edge.io/pull/3862/commits/3b63cc97afd4db4804adedcf1eab733d3cfcbdfc)
